### DISCUSSION TITLE
Simplify logging used in pytest

### DISCUSSION THREAD

**Discussion by JarClass** [pytest-dev/pytest#11127] [Q&A]
I wanted to simplify my current logging system for my program testing. I'm using pytest and python 3.10.8 default logging module. I'm not sure if this is even possible from what I've seen online. Thought I'd ask in case anyone knows if this is possible.
Basically I run the pytest command and while the test is run, its also outputting to a log file that I've created at "../logs/test.log"

**pyproject.toml:**

    [tool.pytest.ini_options]
    log_cli = true
    log_cli_level = "CRITICAL"
    log_cli_format = "%(messages)s"
    log_file = "../logs/test.log"
    log_file_level = "INFO"
    log_file_format = "\n%(asctime)s: [%(levelname)s]\n\t%(message)s (%(filename)s:%(lineno)d)"
    log_file_date_format = "%Y-%m-%d @ %H:%M:%S %Z"

----------------------------------------------------------------

**What's used in the code:**

    import logging
    
    def test_common_count(setUp):
        logging.info("MOST COMMON WORDS AND COUNTS TEST: Using the the test sting. Test string is: %s.", setUp['test_string'])
        text = combine(setUp['test_string'])
        assert text == setUp['expected_string']
        try:
            ... Something something something python code ...
            logging.info("MOST COMMON WORDS AND COUNTS TEST: Result element types and values are as espected. Result is: %s.", result)
        except:
            logging.error("MOST COMMON WORDS AND COUNTS TEST: Result was incorrect. Result is: %s.", result)
            fail()

----------------------------------------------------------------

**Current log file created:**

    2023-06-21 @ 11:46:08 EDT: [INFO]
    	Uploading a txt file (tests_app.py:19)
    
    2023-06-21 @ 11:46:08 EDT: [INFO]
    	Text file name: test.txt (fake_app.py:29)
    
    2023-06-21 @ 11:46:08 EDT: [INFO]
    	Uploading a PDF file (tests_app.py:32)
    
    2023-06-21 @ 11:46:08 EDT: [INFO]
    	PDF file name: test.pdf (fake_app.py:37)
    
    2023-06-21 @ 11:57:52 EDT: [INFO]
    	HYPHEN REMOVAL TEST: Using the the test sting. Test string is: The dog- -s ran down the road to see the other dogs. (tests_manipulate.py:22)
    
    2023-06-21 @ 11:57:52 EDT: [INFO]
    	HYPHEN REMOVAL TEST: Result is as expected. Result is: The dogs ran down the road to see the other dogs. (tests_manipulate.py:27)
    
    2023-06-21 @ 11:57:52 EDT: [INFO]
    	MOST COMMON WORDS AND COUNTS TEST: Using the the test sting. Test string is: The dog- -s ran down the road to see the other dogs. (tests_manipulate.py:33)
    
    2023-06-21 @ 11:57:52 EDT: [INFO]
    	MOST COMMON WORDS AND COUNTS TEST: Result element types and values are as espected. Result is: [('dogs', 2), ('road', 1)]. (tests_manipulate.py:43)

----------------------------------------------------------------

**Desired log file:**

    2023-06-21 @ 11:46:08 EDT: [INFO]
    	Uploading a txt file (tests_app.py:19)
    
    2023-06-21 @ 11:46:08 EDT: [INFO]
    	Uploaded text file: test.txt (fake_app.py:29)
    
    2023-06-21 @ 11:46:08 EDT: [INFO]
    	Uploading a PDF file (tests_app.py:32)
    
    2023-06-21 @ 11:46:08 EDT: [INFO]
    	Uploaded pdf file: test.pdf (fake_app.py:37)
    
    ================================= App tests finished =================================
    
    2023-06-21 @ 11:57:52 EDT: [INFO]
    	HYPHEN REMOVAL TEST: Using the the test sting. Test string is: The dog- -s ran down the road to see the other dogs. (tests_manipulate.py:22)
    
    2023-06-21 @ 11:57:52 EDT: [INFO]
    	HYPHEN REMOVAL TEST: Result is as expected. Result is: The dogs ran down the road to see the other dogs. (tests_manipulate.py:27)
    
    2023-06-21 @ 11:57:52 EDT: [INFO]
    	MOST COMMON WORDS AND COUNTS TEST: Using the the test sting. Test string is: The dog- -s ran down the road to see the other dogs. (tests_manipulate.py:33)
    
    2023-06-21 @ 11:57:52 EDT: [INFO]
    	MOST COMMON WORDS AND COUNTS TEST: Result element types and values are as espected. Result is: [('dogs', 2), ('road', 1)]. (tests_manipulate.py:43)
    
    ================================= Manipulation tests finished =================================

----------------------------------------------------------------
I don't think it would be relevant to this question but i'm including it incase it's helpful.

**Pytest session:**

**command:**
    python -m pytest tests_*.py -s

    ==================================================================================== test session starts ====================================================================================
    platform linux -- Python 3.10.8, pytest-7.3.2, pluggy-1.0.0
    rootdir: /usr/home/Desktop/Projects/extractor
    configfile: pyproject.toml
    plugins: instafail-0.5.0
    collected 7 items                                                                                                                                                                          
    
    tests_app.py::test_upload_textFile PASSED
    tests_app.py::test_upload_pdfFile PASSED
    tests_app.py::test_upload_text PASSED
    tests_app.py::test_fileContentType_fail PASSED
    tests_app.py::test_fileUpload_fail PASSED
    tests_manipulate.py::test_remove_hyphen PASSED
    tests_manipulate.py::test_common_count PASSED
    
    ==================================================================================== 7 passed in 5.35s ====================================================================================

**Comment by simon-liebehenschel** [comment]
Maybe I overlooked something, but could you please clarify what exactly should be "simplified"?

**Comment by Stefanhg** [comment]
Is the problem that you want all output in one start of fixtures to end of all tests + fixtures?
If so then I do agree, this has been annoying me a lot. It is a wrong flow to have in some cases and makes it harder to debug.
